app/Dockerfile

```diff
@@ -5,24 +5,28 @@ RUN mkdir /app /config
 WORKDIR /app
 
 # Install needed packages
-RUN apt-get update
-RUN apt-get install -y python3-virtualenv python3-pip default-mysql-client
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    python3-virtualenv \
+    python3-pip \
+    python3-dev \
+    default-mysql-client \
+    && rm -rf /var/lib/apt/lists/*
```

Comment:
All looks good, but not sure we need this. Looks like its already being deleted on line 28?

Reply:
Oh, it just occured to me that the reason the `rm -rf` command isn't doing anything is because that `/var/lib/{apt,dpkg,cache,log}` expanson is a bash expansion syntax, and dockerfiles are run as `sh`.

```shell
$ bash -c "ls /var/lib/{apt,dpkg,cache,log}"
ls: cannot access '/var/lib/cache': No such file or directory
ls: cannot access '/var/lib/log': No such file or directory
/var/lib/apt:
cdroms.list  daily_lock  extended_states  lists  mirrors  periodic

/var/lib/dpkg:
alternatives  arch  available  cmethopt  diversions  diversions-old  info  lock  lock-frontend	parts  statoverride  statoverride-old  status  status-old  triggers  updates
```
```shell
$ sh -c "ls /var/lib/{apt,dpkg,cache,log}"
ls: cannot access '/var/lib/{apt,dpkg,cache,log}': No such file or directory
```

Reply:
That's really interesting - I've never used dive, thanks for that. Happy to keep in the earlier rm in that case.

Would you mind fixing the rm -rf at the end to what you suggested? Then I'll approve and merge this.

Thanks,
Dan